Source/DafnyCore/Resolver/Resolver.cs

```diff
@@ -9742,7 +9742,22 @@ public class ResolveTypeReturn {
           ExprRhs rr = (ExprRhs)s.Rhs;
           ResolveExpression(rr.Expr, resolutionContext);
           Contract.Assert(rr.Expr.Type != null);  // follows from postcondition of ResolveExpression
-          AddAssignableConstraint(stmt.Tok, lhsType, rr.Expr.Type, "RHS (of type {1}) not assignable to LHS (of type {0})");
+
+          bool isHiddenUpdateToOutVar = false;
+          if (s.Lhs is IdentifierExpr) {
+            IdentifierExpr ie = (IdentifierExpr)s.Lhs;
+            if (ie.Var is Formal) {
+              Formal f = ((Formal)ie.Var);
+              if (!f.InParam) {
```

Comment:
Can you please make these tests in a single if?
E.g.
`if (s.Lhs is IdentifierExpr {Var: Formal {InParam: false}}) {`

Reply:
Nice!  I wasn't aware that you could do this in C#.